util/nrve-niche-payment-handler.py

```diff
@@ -179,51 +148,94 @@ def do_sc_notify(self, event):
+                    self.send_email(
+                        subject,
+                        self.format_error_message({"Transaction Id": tx_hash}, {"From Address": from_address}, {"NRVE Amount": nrve_amount}, {"Number of Transactions": cursor.rowcount})
+                    )
                     return
 
-                # when a payment is outstanding, it will be recorded with the expected from address, the proper nrveAmount (in "neurons")
-                # and a paymentStatus of 0 which indicates it's pending payment
+                block = self.blockchain.GetHeaderByHeight(event.block_number)
+
+                # when a payment is outstanding, it will be recorded with the expected "from address", the proper
+                # nrveAmount (in "neurons") and a paymentStatus of 0 which indicates it's pending payment
                 sql = ("update `NicheAuctionInvoicePayment`\n"
                        "set transactionId = %s\n"
                        ", transactionDate = from_unixtime(%s)\n"
                        "where fromNeoAddress = %s\n"
                        "and nrveAmount = %s\n"
                        "and paymentStatus = 0\n"
                        "and transactionId is null;")
-                args = (tx_hash, timestamp, from_address, nrve_amount)
+                args = (tx_hash, block.Timestamp, from_address, nrve_amount)
 
                 # Create a new record
                 cursor.execute(sql, args)
 
                 if cursor.rowcount != 1:
-                    self.logger.error('Failed updating payment. Should not be possible since it was already locked for update: %s', event)
+                    subject = 'Failed updating payment. Should not be possible since it was already locked for update'
+                    self.logger.error(subject + ': %s', event)
+                    self.send_email(
+                        subject,
+                        self.format_error_message({"Transaction Id": tx_hash}, {"From Address": from_address}, {"NRVE Amount": nrve_amount})
+                    )
                     return
 
-                # send the NRVE to the payment storage address
-                self.transfer_payment(self.niche_payment_storage_address, nrve_amount)
-
                 self.send_email("Successful Niche Payment", log)
 
-            # connection is not autocommit by default. So you must commit to save
-            # your changes.
+            # connection is not autocommit by default. So you must commit to save your changes.
             connection.commit()
+
         except MySQLError as e:
             error_message = 'ERROR: event %s: {!r}, errno is {}'.format(event, e, e.args[0])
             self.logger.error(error_message)
             self.send_email('Niche Payment Error', error_message)
+
         finally:
             connection.close()
 
+    def validate_transaction(self, connection, event, from_address, nrve_amount, tx_hash):
```

Comment:
I wonder if `validate_transaction` is really the right name. It's not really validating anything, right? It's just "handling" an unidentified transaction, so maybe something like `handle_unknow_transaction`?

Reply:
Yeah, I can see that.